### PR TITLE
[DOCS] Replace `twitter` dataset in API conventions + README

### PR DESCRIPTION
**Pull request by jrodewig**
Updates the README and API conventions to use `my-index-*`
in place of the `twitter` dataset.

Also streamlines the README for style and common use cases
(e.g., using autogenerated doc IDs).

### CODE REVIEW

File: README.asciidoc [hunk 1]
```diff
@@ -35,154 +35,148 @@ First of all, DON'T PANIC. It will take 5 minutes to get the gist of what Elasti
...
+can also use separate indices to store different types of data.
 
-Another way to define our simple twitter system is to have a different index per user (note, though that each index has an overhead). Here is the indexing curl's in this case:
+You can configure each index differently. The following request
+creates `my-index-000002` with two primary shards rather than the default of
+one. This may be helpful for larger indices.
 
 ----
-curl -XPUT 'http://localhost:9200/kimchy/_doc/1?pretty' -H 'Content-Type: application/json' -d '
+curl -X PUT http://localhost:9200/my-index-000002?pretty -H 'Content-Type: application/json' -d '
```
Comment (lockewritesdocs) [inline README.asciidoc:135]:
```suggestion
curl -X PUT 'http://localhost:9200/my-index-000002?pretty' -H 'Content-Type: application/json' -d '
```

Comment (lockewritesdocs) [reply README.asciidoc:135]: Single quotes were removed but should be added back.

**Review by lockewritesdocs — APPROVED** [review]
One change to add single quotes back in an example. LGTM otherwise 👍

File: README.asciidoc (hunk 1 above)
Comment (jrodewig) [reply README.asciidoc:135]: Good catch. Thanks.

**Pull request merged**